components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/dao/BPSProfileDAO.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.keyrotation.dao;
+
+import org.apache.log4j.Logger;
+import org.wso2.carbon.identity.keyrotation.config.KeyRotationConfig;
+import org.wso2.carbon.identity.keyrotation.model.BPSPassword;
+import org.wso2.carbon.identity.keyrotation.util.KeyRotationException;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class holds implementations needed to re-encrypt the BPS Profile data in DB.
+ */
+public class BPSProfileDAO {
+
+    private static final Logger log = Logger.getLogger(BPSProfileDAO.class);
+    private static final BPSProfileDAO instance = new BPSProfileDAO();
+    private static final String PROFILE_NAME = "PROFILE_NAME";
+    private static final String USERNAME = "USERNAME";
+    private static final String TENANT_ID = "TENANT_ID";
+    private static final String PASSWORD = "PASSWORD";
+    public static int updateCount = 0;
+    public static int failedCount = 0;
+
+    public BPSProfileDAO() {
+
+    }
+
+    public static BPSProfileDAO getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * To retrieve the list of data in WF_BPS_PROFILE as chunks.
+     *
+     * @param startIndex        The start index of the data chunk.
+     * @param keyRotationConfig Configuration data needed to perform the task.
+     * @return List comprising of the records in the table.
+     * @throws KeyRotationException Exception thrown if something unexpected happens during key rotation.
```

Comment:
It would be better if this can be added descriptively to say the reason for the exception

Reply:
Addressed with https://github.com/wso2/identity-tools/pull/13/commits/6b6184cfde572cff05cdeefedb6e9233fd11d0e8